samples: openthread: add GPIO diag configuration for nRF52840 dongle (#11352)
nrfconnect/sdk-nrf: samples/openthread/coprocessor/README.rst

    |thread_hwfc_enabled|
    In addition, the Co-processor sample reconfigures the baud rate to 1000000 bit/s by default.

Diagnostic module
=================

> The Co-processor sample enables Diagnostic module similarly as described in :ref:`ot_cli_sample_diag_module` section for CLI sample.

Reviewer: ```suggestion
The Co-processor sample enables a diagnostic module in a similar way as described in the :ref:`ot_cli_sample_diag_module` section of the :ref:`ot_cli_sample` sample documentation.
```

> The difference between Co-processor and CLI samples lays in commands execution, which is described in :ref:`ot_coprocessor_testing` section.

Reviewer: ```suggestion
However, the Co-processor and CLI samples use different commands for the module, as described in the :ref:`ot_coprocessor_testing` section.
```

    |thread_hwfc_enabled|
    In addition, the Co-processor sample reconfigures the baud rate to 1000000 bit/s by default.

Diagnostic module
=================

By default, the Co-processor sample comes with the :kconfig:option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER` :ref:`feature set <thread_ug_feature_sets>` enabled, which allows you to use ``diag`` commands.
Use these commands to manually check hardware-related functionalities without running a Thread network.
For example, to ensure radio communication is working when adding a new functionality or during the manufacturing process.

.. note::
    If you disable the :kconfig:option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER` feature set, you can enable the diagnostic module with the :kconfig:option:`CONFIG_OPENTHREAD_DIAG` Kconfig option.

> Additionally for the ``nrf52840dongle_nrf52840`` build target, diagnostic GPIO commands have been enabled by ``openthread_config`` node in :file:`boards/nrf52840dongle_nrf52840.overlay`.

Reviewer 2: Might consider reusing the same section from CLI sample.

Author: It is actually reused, with minor changes as changing sample name and removing
```
See `Testing diagnostic module`_ section for an example.
````
line as this section isn't included in coprocessor sample.
Or do you think that we could include that here too?

Reviewer 2: Lines 85-97 are exactly the same for both samples. My point is that  we should have a single place for edits and in the other place just instantiate the original one.

Author: Ok, added reference to CLI sample with note about difference in command execution

This node has been configured for P0.19 pin, which is connected to reset on nRF52840 Dongle. By setting it in output mode with low state reset is pulled to GND, which results in device rebooting without skipping the bootloader.

> 

Reviewer 3: I'd add some info about why this is useful (we want to enter bootloader from cli and other possible ways to reset from cli fail to do so)

Author: added info in cli sample section
